aws/resource_aws_eip.go

```diff
@@ -324,3 +309,31 @@ func resourceAwsEipDomain(d *schema.ResourceData) string {
 
 	return "standard"
 }
+
+func disassociateEip(d *schema.ResourceData, meta interface{}) error {
+	ec2conn := meta.(*AWSClient).ec2conn
+	log.Printf("[DEBUG] Disassociating EIP: %s", d.Id())
+	var err error
+	switch resourceAwsEipDomain(d) {
+	case "vpc":
+		_, err = ec2conn.DisassociateAddress(&ec2.DisassociateAddressInput{
+			AssociationId: aws.String(d.Get("association_id").(string)),
+		})
+	case "standard":
+		_, err = ec2conn.DisassociateAddress(&ec2.DisassociateAddressInput{
+			PublicIp: aws.String(d.Get("public_ip").(string)),
+		})
+	}
+
+	if err != nil {
```

Comment:
I think the conditional here is superfluous as we're resetting `err` to `nil` below. We can just `return err`